Tests/iaas/image-metadata/image-md-check.py

```diff
@@ -121,17 +145,70 @@ def is_ok(self, props, warn = ""):
 
 
 def is_url(stg):
-    "Is string stg a URL?"
+    """Is string stg a URL?"""
     idx = stg.find("://")
-    if idx < 0:
-        return False
-    if stg[:idx] in ("http", "https", "ftp", "ftps"):
-        return True
-    return False
+    return idx >= 0 and stg[:idx] in ("http", "https", "ftp", "ftps")
+
+
+def parse_date(stg, formats=DATE_FORMATS):
+    """
+    Return time in Unix seconds or 0 if stg is not a valid date.
+    We recognize: %Y-%m-%dT%H:%M:%SZ, %Y-%m-%d %H:%M[:%S], and %Y-%m-%d
+    """
+    bdate = 0
+    for fmt in formats:
+        try:
+            tmdate = time.strptime(stg, fmt)
+            bdate = calendar.timegm(tmdate)
+            break
+        except ValueError:  # as exc:
+            # print(f'date {stg} does not match {fmt}\n{exc}', file=sys.stderr)
+            pass
+    return bdate
+
+
+def freq2secs(stg):
+    """Convert frequency to seconds (round up a bit), return 0 if not applicable"""
+    secs = FREQ_TO_SEC.get(stg)
+    if secs is None:
+        print(f'ERROR: replace frequency {stg}?', file=sys.stderr)
+        secs = 0
+    return secs
+
+
+OUTDATED_IMAGES = []
```

Comment:
Since I have an aversion to mutable globals: I would move this variable into `main` as a local, and then pass it to the relevant functions. Parameters are passed by value, but the value in this case would be the list object (the envelope, if you will) which is itself mutable.

Reply:
I guess that was not quite right. The value is the reference to the list object, and the list object is mutable.